internal/notifier/client.go

```diff
@@ -32,29 +32,90 @@ import (
 	"github.com/hashicorp/go-retryablehttp"
 )
 
-type requestOptFunc func(*retryablehttp.Request)
+type postOption struct {
+	proxy             string
+	certPool          *x509.CertPool
+	requestModifiers  []requestModifier
```

Comment:
I don't think this option needs to be a slice, all the callers are passing only a single function, and that really seems to be enough, I don't see why we would pass more than one function for this